CovertInfiltration/Src/CovertInfiltration/Classes/XComGameState_CovertActionExpirationManager.uc

```diff
@@ -0,0 +1,118 @@
+//---------------------------------------------------------------------------------------
+//  AUTHOR:  statusNone
+//  PURPOSE: Manager class to keep track of covert infiltration expirations
+//---------------------------------------------------------------------------------------
+//  WOTCStrategyOverhaul Team
+//---------------------------------------------------------------------------------------
+
+class XComGameState_CovertActionExpirationManager extends XComGameState_BaseObject;
+
+struct ExpiringAction
+{
+	var() StateObjectReference	ActionRef;
+	var() TDateTime				Expiration;
+	var() TDateTime				OriginTime;
+};
+var() array<ExpiringAction>		ExpiringActions;
+
+static function CreateExpirationManager(optional XComGameState StartState)
+{
+	local XComGameState NewGameState;
+
+	// create a manager on new campaign
+	if (StartState != none)
+	{
+		XComGameState_CovertActionExpirationManager(StartState.CreateNewStateObject(class'XComGameState_CovertActionExpirationManager'));
+	}
+	// seems this never runs cuz reasons
+	if (GetExpirationManager(true) == none)
+	{
+		NewGameState = class'XComGameStateContext_ChangeContainer'.static.CreateChangeState("CI: Creating expiration manager singleton");
+
+		XComGameState_CovertActionExpirationManager(NewGameState.CreateNewStateObject(class'XComGameState_CovertActionExpirationManager'));
+
+		`XCOMHISTORY.AddGameStateToHistory(NewGameState);
+	}
+}
+
+static function Update()
+{
+	local XComGameState_CovertActionExpirationManager ActionExpirationManager;
+	local XComGameState_CovertAction CovertAction;
+	local XComGameState NewGameState;
+
+	local ExpiringAction CurrentAction;
```

Comment:
In same vein as comment about struct name, please rename this variable (to `ExpirationInfo`) as it is super confusing (`CurrentAction` vs `CovertAction`)